micro-infra-camel/src/main/java/com/ofg/infrastructure/camel/CorrelationIdInterceptor.java

```diff
@@ -0,0 +1,59 @@
+package com.ofg.infrastructure.camel;
+
+import com.ofg.infrastructure.correlationid.CorrelationIdUpdater;
+import com.ofg.infrastructure.correlationid.UuidGenerator;
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.apache.camel.Processor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+
+import static com.ofg.infrastructure.correlationid.CorrelationIdHolder.CORRELATION_ID_HEADER;
+
+/**
+ * Interceptor class that ensures the correlationId header is present in {@Exchange}.
+ */
+public class CorrelationIdInterceptor implements Processor {
+
+    private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
```

Comment:
@szpak It would be better, if you review https://github.com/4finance/micro-infra-spring/pull/421 :)

Albeit you are probably right.